### DISCUSSION TITLE
Getting OIDC to work in basic Docker deployment

### DISCUSSION THREAD

**Discussion by fostermi** [drakkan/sftpgo#1594] [General]
Hello,

I'm testing sftpgo for potential adoption, and am trying to get OIDC to work using Docker image drakkan/sftpgo:v2.5. I've created an exact copy of the basic config file, and added the oidc section as described in the documentation. However, when I navigate to the login page, there is "Login with OpenID" button.

The logs show its reading the config file from a bind mount:
```
{"level":"debug","time":"2024-04-18T22:58:56.058","sender":"config","message":"config file used: '\"/var/lib/sftpgo/.config/sftpgo/sftpgo.json\"', config loaded:
```
However, the logs also show that the oidc config looks to empty:
```
OIDC:{ClientID: ClientSecret: ConfigURL: RedirectBaseURL: UsernameField: RoleField: ImplicitRoles:false Scopes:[openid profile email] CustomFields:[] InsecureSkipSignatureCheck:false Debug:false provider:<nil> verifier:<nil> providerLogoutURL: oauth2Config:<nil>}
```
I've even tried to copy that section into the `/etc/sftpgo/sftpg.json` file, but no luck there either.

Am I missing a config somewhere to activate OIDC logins? 

Any help would be appreciated. Thanks.

**Comment by drakkan** [comment]
Hi, try configuring OIDC via env vars, it's simpler because you don't have to understand if you're editing the right configuration file, even if it's quite clear from the logs which configuration file is used

**Comment by fostermi** [comment]
Ok, using ENV vars works. At the moment for testing its easier to use config files, but the one in the repo seems to be broken.

If I copy/paste the sftpgo.json from  [here](https://github.com/drakkan/sftpgo/blob/main/sftpgo.json) and bind mounting to /var/lib/sftpg/.config/sftpgo, sftpgo won't start with the error
```
{"level":"warn","time":"2024-04-19T20:07:41.527","sender":"config","message":"error parsing configuration file: 2 error(s) decoding:\n\n* 'httpd.bindings[0].branding.web_admin.default_css' expected type 'string', got unconvertible type '[]interface {}', value: '[]'\n* 'httpd.bindings[0].branding.web_client.default_css' expected type 'string', got unconvertible type '[]interface {}', value: '[]'"}
```
Looks like https://github.com/drakkan/sftpgo/blob/950cf67e4c03a12c7e439802cabbb0b42d4ee5f5/sftpgo.json#L323 needs to be a string instead of a list.

Same for https://github.com/drakkan/sftpgo/blob/950cf67e4c03a12c7e439802cabbb0b42d4ee5f5/sftpgo.json#L334.

Also, if you set `SFTPGO_CONFIG_DIR=` to any other path than `/etc/sftpgo/`, you'll get this message:

```
{"level":"debug","time":"2024-04-19T20:14:13.617","sender":"dataprovider_sqlite","message":"sqlite database handle created, connection string: \"file:/var/lib/sftpgo/.config/sftpgo/sftpgo.db?cache=shared&_foreign_keys=1\""}
{"level":"error","time":"2024-04-19T20:14:13.617","sender":"dataprovider_sqlite","message":"error preparing database query \"SELECT version from schema_version LIMIT 1\": unable to open database file: no such file or directory"}
```
Its trying to create a database file in the bind mount config directory for some reason.
